src/actions/localisationActions.js

```diff
@@ -0,0 +1,349 @@
+  setSessionLanguageAction,
+} from 'actions/sessionActions';
+import { setAppLanguageAction } from 'actions/appSettingsActions';
+
+import Toast from 'components/Toast';
+
+import type { TranslationData, TranslationResourcesOfLanguage } from 'models/Translations';
+import type { CacheMap } from 'reducers/cacheReducer';
+import type { Dispatch, GetState } from 'reducers/rootReducer';
+
+import { reportLog } from 'utils/common';
+import { getCachedJSONFile } from 'utils/cache';
+
+
+const LOCAL = 'LOCAL';
+const LANGUAGE_ERROR = {
+  MISSES_NAMESPACES: 'Fallback language misses namespaces',
+  NO_TRANSLATIONS: 'Fallback languages has no resources',
+};
+
+type SetLngAndBundle = {
+  language: string,
+  resources: TranslationResourcesOfLanguage,
+  onSuccess?: () => void,
+}
+
+const getTranslationData = (lng: string) => {
+  return localeConfig.namespaces.map((ns) => ({
+    ns,
+    url: `${localeConfig.baseUrl}${lng}/${ns}.json`, // eslint-disable-line i18next/no-literal-string
+  }));
+};
+
+const getCachedTranslationResources =
+  async (translationsData: TranslationData[], cacheMap: CacheMap, dispatch: Dispatch) => {
+    const cachedTranslations = await Promise.all(translationsData.map(async ({ ns, url }) => {
+      const { localUrl } = cacheMap?.[url] || {};
+      if (!localUrl) return { ns, translations: {} };
+
+      const translations = await getCachedJSONFile(localUrl);
+      if (!translations) {
+        // cached file no longer exists - remove it from map
+        dispatch(removeUrlCacheAction(url));
+        return { ns, translations: {} };
+      }
+      return { ns, translations };
+    }));
+
+    return cachedTranslations.reduce((formattedResources, translation) => {
+      const { ns, translations } = translation;
+      if (ns && translations) formattedResources[ns] = translations;
+      return formattedResources;
+    }, {});
+  };
+
+const getTranslationsResources = async (props) => {
+  const {
+    language,
+    dispatch,
+    getState,
+  } = props;
+  let resources;
+  // TODO: pass in versioning;
+  let version = '';
+  const missingNsArray = [];
+  const translationsData = getTranslationData(language);
+
+  const { session: { data: { isOnline } } } = getState();
+
+  const relatedLocalTranslationData = localeConfig.localTranslations[language];
+
+  // if translations' baseUrl is provided - use external translations. If not - local.
+  if (localeConfig.baseUrl) {
+    // If network is available - fetch and cache newest translations
+    // TODO: decide on how frequent to update translations - now its fetching newest all the time
+    if (isOnline) {
+      // fetches to storage and set local path to cacheMap
+      await Promise.all(translationsData.map(({ url }) => dispatch(cacheUrlAction(url))));
+    }
+
+    // get newest cached translations
+    const { cache: { cacheMap } } = getState();
+    resources = await getCachedTranslationResources(translationsData, cacheMap, dispatch);
```

Comment:
If anything fails in `getCachedTranslationResources`, empty object will be returned.
And if resources are empty, locally used translations are added.
```
    // check missing namespaces
    const existingNameSpaces = Object.keys(resources).filter(ns => !isEmpty(resources[ns]));
    const missingNameSpaces = localeConfig.namespaces.filter(ns => !existingNameSpaces.includes(ns));
    if (missingNameSpaces.length) {
      // found missing name spaces - add locally stored ones (if any)
      // const relatedLocalTranslationData = localeConfig.localTranslations[language];
      if (relatedLocalTranslationData) {
        version = LOCAL;
        const missingTranslationResources = missingNameSpaces.reduce((allResources, ns) => {
          const relatedTranslations = relatedLocalTranslationData[ns];
          if (relatedTranslations) {
            allResources[ns] = relatedTranslations;
          } else {
            missingNsArray.push(ns);
          }
          return allResources;
        }, {});
        resources = { ...resources, ...missingTranslationResources };
      }
    }
```

Now, if it still has missing namespaces (compared to the ones in config), I'm reporting it to Sentry. 
And also informing users that some translations might be missing.
And in case this happens, I'm always adding default language translations as resource bundles to `i18next` and they are then used as fallbacks by design.

Reply:
Thank you for the explanation 👍